src/app/modules/crypto/sagas.ts

```diff
@@ -42,7 +42,40 @@ export function* requirePassword(): SagaIterator {
   }
 }
 
+export function* changePassword(): SagaIterator {
+  try {
+    yield call(requirePassword);
+    
+    // temporarily hold onto the previous salt in case the password is 
+    // not reset and we need to restore it
+    const prevSalt = yield select(selectSalt);
+    // generate a new salt since we are expecting a new password
+    yield put(generateSalt());
+
+    yield put({ type: types.CHANGING_PASSWORD });
+    const action = yield take([ types.CANCEL_CHANGE_PASSWORD, types.SET_PASSWORD]);
+    if (action.type === types.CANCEL_CHANGE_PASSWORD) {
+      // revert to the previous salt since the password was not changed
+      yield put({
+        type: types.GENERATE_SALT,
+        payload: prevSalt,
+      })
+      return;
+    }
+
+    // clear the saves password cache since it's no longer valid
+    yield call(clearPasswordCache);
+    yield put({ type: types.CHANGE_PASSWORD_SUCCESS });
```

Comment:
I originally had it saving to cache, then realized that we'd want to first check if a password was already cached since the user may have unchecked the `Don't ask me again this session` checkbox. I don't think its good to cache the new password if the user didn't want the old one cached. Implementing this conditional logic added a bit more code to the saga that I wasn't sure was necessary. I'll add it back and get your thoughts on whether or not it's worth it.

Reply:
Ah, that's a good thought! Should be a pretty small implementation though,
```ts
cachedPassword = yield call(getPasswordCache);
yield put(enterPassword({
  password: newPassword,
  save: !!cachedPassword,
});
```

Reply:
Well, that's far simpler than what I had in mind 😆 . Thanks for the code snippet. I'll use that.

Reply:
I added this code to cache the new password